### PR TITLE
4PL Basic Model (no guessing)

### PR DESCRIPTION
**Pull request by EntilZha**
Implement initial 4PL IRT model, it has feasibility, but not guessing yet.

Unit test could be improved later, but I_think creating a common test across all models is a better way to do that.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -19,6 +19,11 @@ pytest = "^6.2.4"
 pytest-cov = "^2.12.1"
 pylint = "^2.8.3"
 black = "^21.5b2"
+isort = "^5.8.0"
+ipython = "^7.24.1"
+
```
Comment (jplalor) [inline pyproject.toml:24]: Can you add rich here? It's causing the build to fail.

Comment (EntilZha) [reply pyproject.toml:24]: I added a requirements file generated from poetry with `poetry export -f requirements.txt --output requirements.txt` and have CI set to use that, hopefully that fixes the issue

**Comment by EntilZha** [comment]
Think the issue you pointed out is now fixed

**Comment by jplalor** [comment]
Yep looks like it. Thanks!

**Pull request merged**